appmap/_implementation/testing_framework.py

```diff
@@ -124,4 +126,11 @@ def record(self, klass, method, **kwds):
         filename = fi.filename + '.appmap.json'
         metadata = fi.metadata
         metadata.update(self.metadata)
-        (self.appmap_path / filename).write_text(generation.dump(rec, metadata))
+        appmap_json = self.appmap_path / filename
+        with NamedTemporaryFile(mode='w', dir=self.appmap_path, delete=False) as f:
+            f.write(generation.dump(rec, metadata))
+            try:
+                os.remove(appmap_json)
+            except FileNotFoundError:
+                pass
```

Comment:
This is not necessary; `os.replace()` replaces the destination (as the name implies). What's more, this can lead to race with eg. an editor that has the file open. It could see the file removed and erronously inform the user it has been removed, instead of replaced. `os.replace()` replaces atomically if possible.